### ISSUE TITLE
Slow/stalling navigation from feed to event

### ISSUE THREAD

**Issue by minaorangina** [DRDD2016/native#183]
clicking on a feed item to go to an event, often the view gets stuck

**Comment by minaorangina** [comment]
### Problem:
This is to do with updating the Navigation parameters for use in the Navigation bar.
We display the name of the event/name of the route in the navigation bar.  For cases where the data required for the navigation bar isn't available yet (e.g. requesting data from server). ex-navigation provides an [`updateCurrentRouteParams` method](https://github.com/exponent/ex-navigation/#updating-route-params).

The source of the problem is here:
```js
// src/js/event/index.js

componentDidMount () {
    setTimeout(() => {
      if (this.props.userIsHost) {
        this.props.navigator.updateCurrentRouteParams({
          name: this.props.event.name,
          userIsHost: this.props.userIsHost,
          navigator: this.props.navigator,
          event: this.props.event,
          handleEdit: this.props.handleEdit
        });
      } else {
        this.props.navigator.updateCurrentRouteParams({
          name: this.props.event.name
        });
      }
    }, 300);
  }
```
Running this code (even without the `setTimeout`) prevents the navigation from transitioning smoothly.


### Solution
When pushing to a new route, you can [pass down a params object](https://github.com/exponent/ex-navigation/#passing-params-to-a-route) as an optional second argument.

Decided to directly pass down the params data when initially pushing to the new route, just after fetching the event data.  This way, no need to use `updateCurrentRouteParams`.
```js
dispatch(getEventSuccess(data));
const params = { /* enter params here */ };
pushTo('event', params);
```

**Comment by heron2014** [comment]
Resolved by #185
